### PR TITLE
add reconciler for runtime hook standalone work mode

### LINKED ISSUE #120: [proposal] runtime-hooks should support multiple running mode
**What is your proposal**:
injections in runtime hooks of koordlet such as cpu qos，memory oqs，llc/mba can work with or without runtime-hook-manager（runtime-porxy mode）

**Why is this needed**:
someone would rather use bypass-injection mode than changing kubelet cri endpoint config.

**Is there a suggested solution, if so, please add it**:
runtime hooks in koordlet should support two running mode: injection(with cri proxy)，bypass(without cri proxy)

### PR DESCRIPTION
**Pull request by zwzhang0107**
### Ⅱ. Does this pull request fix one issue?

fixes #120 

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: pkg/koordlet/runtimehooks/protocol/pod_context.go [hunk 1]
```diff
@@ -110,7 +111,7 @@ func (p *PodContext) injectForOrigin() {
 func (p *PodContext) injectForExt() {
 	if p.Response.Resources.CPUBvt != nil {
 		if err := injectCPUBvt(p.Request.CgroupParent, *p.Response.Resources.CPUBvt); err != nil {
-			klog.Infof("set pod %v/%v bvt %v on cgroup parent %v failed, error %v", p.Request.PodMeta.Namespace,
+			klog.Infof("set pod q%v/%v bvt %v on cgroup parent %v failed, error %v", p.Request.PodMeta.Namespace,
```
Comment (eahydra) [inline pkg/koordlet/runtimehooks/protocol/pod_context.go:114]: q%v, typo?

File: pkg/koordlet/runtimehooks/reconciler/reconciler.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	// new watcher will be added with new pod created, and deleted with pod destroyed
+	for {
+		select {
+		case <-c.podUpdated:
+			podsMeta := c.getPodsMeta()
+			for _, podMeta := range podsMeta {
+				for _, r := range globalCgroupReconciler[PodLevel] {
+					podCtx := protocol.HooksProtocolBuilder.Pod(podMeta)
+					if err := r.fn(podCtx); err != nil {
+						klog.Infof("calling reconcile function %v failed, error %v", r.description, err)
```
Comment (eahydra) [inline pkg/koordlet/runtimehooks/reconciler/reconciler.go:160]: klog.Errorf?

File: pkg/koordlet/runtimehooks/reconciler/reconciler.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+			}
+		case <-stopCh:
+			klog.Infof("stop reconcile pod cgroup")
+			return
+		}
+	}
+}
+
+func init() {
+	globalCgroupReconciler = map[ReconcilerLevel]map[string]*cgroupReconciler{
```
Comment (eahydra) [inline pkg/koordlet/runtimehooks/reconciler/reconciler.go:189]: It is simpler to initialize directly when declaring variable X.

**Review by jasonliu747 — CHANGES_REQUESTED** [review]
Hi please set the logging level correctly according to our [conventions](https://github.com/koordinator-sh/koordinator/blob/main/docs/contributors/logging.md).

File: pkg/koordlet/runtimehooks/reconciler/reconciler.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	}
+}
+
+func doKubeQOSCgroup() {
+	for _, kubeQOS := range []corev1.PodQOSClass{
+		corev1.PodQOSGuaranteed, corev1.PodQOSBurstable, corev1.PodQOSBestEffort} {
+		for _, r := range globalCgroupReconciler[KubeQOSLevel] {
+			kubeQOSCtx := protocol.HooksProtocolBuilder.KubeQOS(kubeQOS)
+			if err := r.fn(kubeQOSCtx); err != nil {
+				klog.Infof("calling reconcile function %v failed, error %v", r.description, err)
```
[the remark this answers is not in the document]
Comment (zwzhang0107) [reply pkg/koordlet/runtimehooks/reconciler/reconciler.go]: use warning for some procedure failing

**Comment by zwzhang0107** [comment]
> Hi please set the logging level correctly according to our [conventions](https://github.com/koordinator-sh/koordinator/blob/main/docs/contributors/logging.md).

lines updated for this pr

File: pkg/koordlet/runtimehooks/protocol/container_context.go [hunk 5]
```diff
@@ -122,3 +126,12 @@ func (c *ContainerContext) injectForOrigin() {
 func (c *ContainerContext) injectForExt() {
 	// TODO
 }
+
+func getContainerID(podAnnotations map[string]string, containerUID string) string {
+	// TODO parse from runtime hook request directly
+	runtimeType := "containerd"
+	if _, exist := podAnnotations["io.kubernetes.docker.type"]; exist {
```
[the remark this answers is not in the document]
Comment (zwzhang0107) [reply pkg/koordlet/runtimehooks/protocol/container_context.go:133]: not yet, its a temp work around

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Review by jasonliu747 — APPROVED** [review]

**Pull request merged**